src/main/java/org/osiam/auth/login/internal/InternalAuthenticationProvider.java

```diff
@@ -117,13 +113,11 @@ public Authentication authenticate(Authentication authentication) {
             }
         }
 
-        User authUser = new User.Builder(username).setId(user.getId()).build();
-
         List<GrantedAuthority> grantedAuthorities = user.getRoles().stream()
                 .map(role -> new SimpleGrantedAuthority(role.getValue()))
                 .collect(Collectors.toList());
 
-        return new InternalAuthentication(authUser, password, grantedAuthorities);
+        return new InternalAuthentication(username, password, grantedAuthorities);
```

Comment:
Unfortunately, this does not work. The principal must be the `User`, because it's the base for generating the access tokens in the resource owner credentials grant and authorization code grant. I get a bunch of errors when running the ITs related to missing data in the access tokens.